.github/workflows/create-tag.yml

```diff
@@ -0,0 +1,118 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'create-tag'
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: 'The name of the tag to be created.'
+        type: 'string'
+        required: true
+      annotated_tag:
+        description: 'Whether it will be an annotated tag in Git.'
+        type: 'boolean'
+        default: true
+        required: false
+      branch:
+        description: 'The branch to create tag from.'
+        type: 'string'
+        required: false
+        default: 'main'
+      message:
+        description: 'Message for the tag.'
+        type: 'string'
+        required: false
+      deployment_environment:
+        description: 'The environment with protection rules, see reference https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment'
+        type: 'string'
+        required: true
+      token_minter_wif_provider:
+        description: 'The full identifier of the Workload Identity Provider, including the project number, pool name, and provider name.'
+        type: 'string'
+        required: true
+      token_minter_wif_service_account:
+        description: 'Email address or unique identifier of the Google Cloud service account for which to generate credentials.'
+        type: 'string'
+        required: true
+      token_minter_service_audience:
+        description: 'Cloud Run audience for the production github-token-minter service'
+        type: 'string'
+        required: true
+      token_minter_service_url:
+        description: 'URL for the production github-token-minter service'
+        type: 'string'
+        required: true
+
+jobs:
+  create-tag:
+    runs-on: 'ubuntu-latest'
+    environment: '${{ inputs.deployment_environment }}'
+    steps:
+      - name: 'Mint Token'
+        id: 'mint-token'
+        uses: 'abcxyz/github-token-minter/.github/actions/mint-token@af661f898c6b0cda46dcaadd8299fb1eab264975' # ratchet:abcxyz/github-token-minter/.github/actions/mint-token@main
+        with:
+          wif_provider: '${{ inputs.token_minter_wif_provider }}'
+          wif_service_account: '${{ inputs.token_minter_wif_service_account }}'
+          service_audience: '${{ inputs.token_minter_service_audience }}'
+          service_url: '${{ inputs.token_minter_service_url }}'
+          requested_permissions: |-
+            {
+              "repositories": ["${{ github.event.repository.name }}"],
+              "permissions": {
+                "contents": "write"
+              }
+            }
+      - name: 'Create Tag'
+        uses: 'actions/github-script@98814c53be79b1d30f795b907e553d8679345975' # ratchet:actions/github-script@v6
+        with:
+          github-token: '${{ steps.mint-token.outputs.token }}'
+          retries: '3'
+          script: |+
+            let sha = '${{ github.sha }}'
+            if ('${{ inputs.branch }}' !== 'main') {
```

Comment:
We should instead check against the default branch instead of main --> `${{ github.event.repository.default_branch }}`

Reply:
on a second thought, we should check against `[github.ref_name](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)` instead. Here is to get the sha of the requested branch (to create tags on). if the requested branch is the same branch as the event ref_name, I can just use `github.sha`

Reply:
i dont think that is the case, we want the workflow to be run from main right, but the target branch might be something else

Reply:
I could be missing something, but my understanding is that the workflow branch != input branch. If you are talking about the branch where the workflow is from, We can use deployment protected/selected branches to ensure it is allowlisted by us.
<img width="361" alt="Screenshot 2023-09-29 at 1 53 05 PM" src="https://github.com/abcxyz/pkg/assets/51539171/229f9d6c-856c-4dea-8d60-f2657ed1e660">

Reply:
I tried to use workflow from a different branch (tmp) [here](https://github.com/sqin2019/hello-world/actions/runs/6357165921/job/17267936647), `github.ref_name == workflow branch (tmp here)`.  Do you mean sth else with target branch?

Reply:
I think what im saying is if i create a workflow on a random branch and then say hey run this workflow from my branch, its not reviewed. So we should make sure the workflow is run from main (or the default branch), always. The input branch is the branch in which we create the tag on, this does not necessarily mean that the workflow branch and the input branch are equal. We can chat monday if needed.

Reply:
I see, we should really add a condition on the create-tag job, so we only run the job when `${{ github.ref_name }} === ${{ github.event.repository.default_branch }}

Reply:
Why should we allow non default_branch?

Reply:
If you mean why allow tags on non default_branch, as Mike mentioned we might have a dedicated branch for release which is not necessarily the default branch.

Reply:
I actually think it's a better idea to only allow this input once we really have a release branch. For one, our immediate goal is to avoid creating tags on non default branch (which is our only release branch). Having it as an input before we can enforce multi-party review is problematic. For two, we haven't discussed how release branches should work so introducing this input is a bit premature.

Reply:
I see, for the interim, we can skip the job when the input branch is not default branch. updating, thx!